examples/crn-parser/main.tf

```diff
@@ -14,11 +14,16 @@ module "resource_group" {
 # COS instance
 ########################################################################################################################
 
-resource "ibm_resource_instance" "cos_instance" {
-  name              = "${var.prefix}-cos"
+module "key_protect" {
+  source            = "terraform-ibm-modules/key-protect/ibm"
+  version           = "2.8.4"
   resource_group_id = module.resource_group.resource_group_id
-  service           = "cloud-object-storage"
-  plan              = "standard"
-  location          = "global"
-  tags              = var.resource_tags
+  region            = var.region
+  key_protect_name  = "${var.prefix}-kp"
+  access_tags       = var.resource_tags
+}
```

Comment:
We really don't need to provision key protect instance here, or a resource group. It kind of doesn't make sense as a real use case, because that module already supports outputting region, guid etc, so you wouldn't use the CRN parser here.
I suggest that the example just takes in a CRN as a single input variable. In the test we can then pass it a mock CRN value. (it means updating the test to use `TestOptionsDefault` instead of `TestOptionsDefaultWithVars`)